### ISSUE TITLE
App re register failed ( dont jump to conclusions )

### ISSUE THREAD

**Issue by Kai-Denzel-Jane** [MCMrARM/mc-w10-version-launcher#260]
Hello before I say anything yes I've tried #246  and so on but from the error below I noticed it can't find Microsoft.Services.Store.Engagement, and obviously requires it to continue.

![image](https://github.com/MCMrARM/mc-w10-version-launcher/assets/120575219/cbc7cb38-3685-4682-9410-d13d4837f4de)

**Comment by Kai-Denzel-Jane** [comment]
Might I add that yes I have the Visual C++ Installed and developer mode is in fact enabled.

Also when trying to launch the versions .exe itself it complains about missing .dll files which have Visual C++ naming patterns but like I said I have the Visual C++ installed and have tried reinstalling but nothing has changed so far.

**Comment by Kai-Denzel-Jane** [comment]
System.Exception: Deployment failed: Windows cannot install package Microsoft.MinecraftUWP_1.19.8301.0_x64__8wekyb3d8bbwe because this package depends on a framework that could not be found. Provide the framework "Microsoft.Services.Store.Engagement" published by "CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US", with neutral or x64 processor architecture and minimum version 10.0.1610.0, along with this package to install. The frameworks with name "Microsoft.Services.Store.Engagement" currently installed are: {Microsoft.Services.Store.Engagement_10.0.19011.0_x64__8wekyb3d8bbwe Microsoft.Services.Store.Engagement_10.0.19011.0_x86__8wekyb3d8bbwe}

This is the error from the log

**Comment by Kai-Denzel-Jane** [comment]
So its saying I need 10.0.1610.0 minimum even though I clearly have 10.0.19011.0 installed

**Comment by Retapich** [comment]
You either installed the required packages incorrectly, or you have older versions of the required packages installed.

I was able to resolve this (_and one other_) error in **Win10 LTSC 2019 (v1809)** by installing the following latest version packages:
- Microsoft.Services.Store.Engagement_8wekyb3d8bbwe
- Microsoft.VCLibs.140.00_8wekyb3d8bbwe

The versions in the package names were removed to make it easy to download the latest version from the "[store.rg-adguard.net](https://store.rg-adguard.net)" site by copying the package name into the site form⤵️
![Screen-700](https://github.com/MCMrARM/mc-w10-version-launcher/assets/19671942/3f148c5b-60d7-4018-9c38-d28a4ddeeae4)

It is enough to install only the **x64** version of the packages (**ofc, if you have a 64-bit OS**) - downloading through the context menu, because **Chromium**-based browsers do not want to download one click on the hyperlink itself⤵️
![Screen-701](https://github.com/MCMrARM/mc-w10-version-launcher/assets/19671942/b362f566-ba7a-4de4-b196-d46e4f80ae69)
You will also need to refresh (_press F5_) this newly opened tab for the package download to start.

I install/update the packages using **PowerShell** (_with admin rights; hotkey "Win+X"_), having previously placed the packages in the root of the **"C" drive**. Through the following commands⤵️
```
add-appxpackage C:\Microsoft.VCLibs.140.00_14.0.32530.0_x64__8wekyb3d8bbwe.appx
add-appxpackage C:\Microsoft.Services.Store.Engagement_10.0.19011.0_x64__8wekyb3d8bbwe.appx
```

The version of the packages may change in the future, so be sure to check the name of the packages in the above commands if you copy them completely.
If there is a difference in the name, you can erase all the characters after the first underscore ↘️ _ ↙️ and press the "**Tab**" key to automatically fill in the actual package name.

In addition, if you are like me and use the **LTSC edition** (_or the old "Pro" or "Home" edition - it should not be older than version 1809_), this will not be enough to install the game↙️

You will also need to edit the file "**AppxManifest.xml**" in the folder with the unzipped game files, which will be located in the folder with the file "**MCLauncher.exe**" (only if you downloaded at least some version of Minecraft via "MCLauncher.exe").

> As an example, the latest version of the game is **1.20.0.1** - so the folder with the game files will be called "**Minecraft-1.20.0.1**", and the "**AppxManifest.xml**" file will be located there.

Basically, you need to replace the value of the "**MinVersion**" parameter with "**10.0.17134.0**" - as shown in the screenshot below⤵️
![Screen-702](https://github.com/MCMrARM/mc-w10-version-launcher/assets/19671942/3f162249-8f62-4eaf-9c6f-23a52d255f96)

After that, you can save this file and start the installation of the game through the "**Launch**" button in the launcher itself.
### **✅ Done!**

___________________
Perhaps someone who has read this far and has a deep knowledge of Windows OS will be able to advise something about this [ #110](https://github.com/MCMrARM/mc-w10-version-launcher/issues/110#issuecomment-1599286476)  - It appeared only on one PC out of two with the same Win10 LTSC. I would really appreciate your attention🙏

**Comment by GreatNateDev** [comment]
Impressive write up

**Comment by miqumi** [comment]
> You either installed the required packages incorrectly, or you have older versions of the required packages installed.
> 
> I was able to resolve this (_and one other_) error in **Win10 LTSC 2019 (v1809)** by installing the following latest version packages:
> 
> * Microsoft.Services.Store.Engagement_8wekyb3d8bbwe
> * Microsoft.VCLibs.140.00_8wekyb3d8bbwe
> 
> The versions in the package names were removed to make it easy to download the latest version from the "[store.rg-adguard.net](https://store.rg-adguard.net)" site by copying the package name into the site form⤵️ ![Screen-700](https://user-images.githubusercontent.com/19671942/247313360-3f148c5b-60d7-4018-9c38-d28a4ddeeae4.png)
> 
> It is enough to install only the **x64** version of the packages (**ofc, if you have a 64-bit OS**) - downloading through the context menu, because **Chromium**-based browsers do not want to download one click on the hyperlink itself⤵️ ![Screen-701](https://user-images.githubusercontent.com/19671942/247313384-b362f566-ba7a-4de4-b196-d46e4f80ae69.png) You will also need to refresh (_press F5_) this newly opened tab for the package download to start.
> 
> I install/update the packages using **PowerShell** (_with admin rights; hotkey "Win+X"_), having previously placed the packages in the root of the **"C" drive**. Through the following commands⤵️
> 
> ```
> add-appxpackage C:\Microsoft.VCLibs.140.00_14.0.32530.0_x64__8wekyb3d8bbwe.appx
> add-appxpackage C:\Microsoft.Services.Store.Engagement_10.0.19011.0_x64__8wekyb3d8bbwe.appx
> ```
> 
> The version of the packages may change in the future, so be sure to check the name of the packages in the above commands if you copy them completely. If there is a difference in the name, you can erase all the characters after the first underscore ↘️ _ ↙️ and press the "**Tab**" key to automatically fill in the actual package name.
> 
> In addition, if you are like me and use the **LTSC edition** (_or the old "Pro" or "Home" edition - it should not be older than version 1809_), this will not be enough to install the game↙️
> 
> You will also need to edit the file "**AppxManifest.xml**" in the folder with the unzipped game files, which will be located in the folder with the file "**MCLauncher.exe**" (only if you downloaded at least some version of Minecraft via "MCLauncher.exe").
> 
> > As an example, the latest version of the game is **1.20.0.1** - so the folder with the game files will be called "**Minecraft-1.20.0.1**", and the "**AppxManifest.xml**" file will be located there.
> 
> Basically, you need to replace the value of the "**MinVersion**" parameter with "**10.0.17134.0**" - as shown in the screenshot below⤵️ ![Screen-702](https://user-images.githubusercontent.com/19671942/247314731-3f162249-8f62-4eaf-9c6f-23a52d255f96.png)
> 
> After that, you can save this file and start the installation of the game through the "**Launch**" button in the launcher itself.
> 
> ### **✅ Done!**
> Perhaps someone who has read this far and has a deep knowledge of Windows OS will be able to advise something about this [ #110](https://github.com/MCMrARM/mc-w10-version-launcher/issues/110#issuecomment-1599286476) - It appeared only on one PC out of two with the same Win10 LTSC. I would really appreciate your attention🙏

thanks a bajigillion
